docs/wiki/research/account-abstraction/eip-7702.md

```diff
@@ -7,15 +7,76 @@
+1. Basic validations are performed, such as verifying `chain_id < 2^256`, `len(auth.address) == 20` etc.. Additionally, the `chain_id` must either match the current chain's ID or be set to `0`.
+2. Verify that the `nonce` is less than `2^64 - 1`.
+3. The `authority` is recovered from the signature and the payload using the `ecrecover` method:  
+`authority = ecrecover(msg, y_parity, r, s).` where `msg = keccak(MAGIC || rlp([chain_id, address, nonce]))` and `s` must be less than `secp256k1n/2`.
+4. The recovered `authority` is marked as a warm address and is added to the `accessed_addresses` set.
+5. Verify that `nonce` is equal to  nonce of `authority` if the `authority` exists in the trie otherwise `nonce` has to equal `0`.
+6. If `authority` exists in the trie the refund amount of `PER_EMPTY_ACCOUNT_COST - PER_AUTH_BASE_COST` is added to the global refund mechanism.
+7. In case of a non-zero `address` the code of the authority is set to `0xef0100 || address` (where `||` denotes concatenation), otherwise the EOA's code is cleared and the code hash is set to the hash of a null address (`0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470`). This serves as a revocation mechanism.
+8. The nonce of `authority` is incremented by one.
+
+### Creation by template
+
+Due to space optimization concerns, the traditional `initCode` is replaced by a "code pointer" mechanism, with initialization occurs during a regular call.
+[EIP-3607](https://eips.ethereum.org/EIPS/eip-3607) specifies that an account with deployed code cannot originate transactions.  
+To work around this the `code` field is populated with a 23-byte value composed of a prefix and the delegated contract’s address.
+The prefix is 3 bytes long:
+ - The first byte (`0xef`) is a banned opcode, as defined in [EIP-3541](https://eips.ethereum.org/EIPS/eip-3541).
+ - The following two bytes (`0x0100`) act as a 7702-specific designator.
+
+### Opcode changes
+
+Some reading and execution instructions are impacted by this design, including:
+  - [EXTCODESIZE](https://www.evm.codes/?fork=cancun#3b) will return 2, the length of `0xef01`
+  - [EXTCODEHASH](https://www.evm.codes/?fork=cancun#3f) will return `keccak256` hash of `0xef01`
+  - [EXTCODECOPY](https://www.evm.codes/?fork=cancun#3c) will return the bytes `0xef01`
+  - Execution instructions (`CALL`, `CALLCODE`, `STATICCALL`, `DELEGATEACLL`) will execute the code from the delegated contract address, but within the context of the EOA
+
+### Gas costs
+
+Each authorization tuple incurs a gas cost of 25,000 gas, calculated using the following formula:
+```21000 + 16 * non-zero calldata bytes + 4 * zero calldata bytes + 1900 * access list storage key count + 2400 * access list address count```    
+This approach charges the maximum gas amount upfront. A partial refund of 12,500 gas is issued if the contract already exists, which happens through the global refund mechanism.
+
```

Comment:
could be nice to tie together all the info above with a simple example, i.e.
Step 1: user submits tsx with `authorization_list` pointing to a SC with e.g. session key logic capabilities
Step 2:***